### ISSUE TITLE
ngspice

### ISSUE THREAD

**Issue by harieamjari** [termux/termux-packages#8143]
**Package description**

ngspice is the open source spice simulator for electric and electronic circuits.

**Link to home page and sources**
1. Home page: http://ngspice.sourceforge.net/
2. Source code: https://sourceforge.net/projects/ngspice/files/ng-spice-rework/31/ngspice-31.tar.gz/download

**Additional information**
Have you compiled or tried to compile the package on device?
* If it didn't work then please provide the error you ran into.
* If it did work then please share any patches that you had to apply.

I have tried compiling ngspice-31 in my android 6.0.1 (architecture armv7l).

There was a slight modification in the source as describe in https://github.com/termux/unstable-packages/pull/122/files. (note that the modification carried in ngspice-30 worked in ngspice-31).

After I installed the following dependencies (see url above), I ran `./autogen.sh && ./configure --enable-xspice --enable-cider --enable-pss --enable-debug=no` and finally `make`.

gcc spewed many warnings like 

```
unknown warning option '-Wno-unused-but-set-variable'; did you mean '-Wno-unused-const-variable'?
      [-Wunknown-warning-option]
```
and it took nearly 5 Hours to compile all. maybe because of my low-end android =(

I did not include `--enable-openmp` in the configure because when running `make`, gcc prints out an error like `cannot find sys/timeb.h` (which is bullcrap; I needed to run `./configure` again with different parameter and wait for 20 minutes). I wonder why?
![Screenshot_20200518-011015](https://user-images.githubusercontent.com/46589135/82155000-6b458c80-98a4-11ea-8c1b-a01ac43b7dca.png)
Picture shows using gnuplot. (note when gnuplot.c in the source code is not modified, it will try to use X11 terminal type (which doesn't exist in my gnuplot). In this one I used dumb terminal type which prints an ascii art for anything that prints text.).

If you need elaboration, feel free to ask~

**Comment by BeegCat** [comment]
@harieamjari hello, I'm trying to accomplish the same goal as you!! Can you help me with the exact processes? Mine gives me a x11 aborted error.

**Comment by harieamjari** [comment]
> @harieamjari hello, I'm trying to accomplish the same goal as you!! Can you help me with the exact processes? Mine gives me a x11 aborted error.
It's been long since I tried compiling it, and the link seems to be not working. Anyway is that a runtime error? I assume it tried to use the X11 terminal (which doesn't exist). In this one I used dumb terminal type which prints an ascii art. I have long forgotten the steps, but since I documented myself, all I can give info is to modify "ngspice-34/src/frontend/plotting/gnuplot.c".
